### PR TITLE
All: Disable WebDAV response caching

### PR DESCRIPTION
**Pull request by roman-r-m**
Fixes https://github.com/laurent22/joplin/issues/4706 and maybe other sync failures too.

The issue reported on Android but I think it makes sense to disable caching everywhere as we need to get the latest data from the server when syncing.

### CODE REVIEW

**Comment by tessus** [comment]
Nice catch.

However, I'm wondering why caching was turned on in the first place. Is it the default? Seems odd to me. Usually caching has to be turned on explicitly, because of the possibility of stale data. Very strange.

**Comment by roman-r-m** [comment]
> Nice catch.
> 
> However, I'm wondering why caching was turned on in the first place. Is it the default? Seems odd to me. Usually caching has to be turned on explicitly, because of the possibility of stale data. Very strange.

Strange indeed, I'll check whether its on Android or on the server (Apache)

**Comment by roman-r-m** [comment]
On Android this is set here
https://github.com/facebook/react-native/blob/9cd98fa3eeded9f9bded846ac8ec94f05052a6c7/ReactAndroid/src/main/java/com/facebook/react/modules/network/OkHttpClientProvider.java#L66-L69

Not sure if it's the same on iOS.
I still think it makes sense to have it set everywhere as an extra precaution even if the issue only occurs on Android.

**Comment by laurent22** [comment]
Indeed I can't see any potential issue with explicitly disabling cache. We request the list of file timestamps anyway, so we know which files we need to download and when we do that we definitely want the latest version.

Since it's share hosting I suspect it's due a buggy cache implementation on their side.

File: packages/lib/WebDavApi.js [hunk 1]
```diff
@@ -352,6 +352,10 @@ class WebDavApi {
 			if (method === 'PUT') headers['Content-Type'] = 'text/plain';
 		}
 
+		if (!headers['Cache-Control']) {
+			headers['Cache-Control'] = 'no-store';
```
Comment (laurent22) [inline packages/lib/WebDavApi.js:356]: Actually since it's a workaround for a server bug, could you please add a comment and link to the original issue?

**Comment by roman-r-m** [comment]
> Since it's share hosting I suspect it's due a buggy cache implementation on their side.

No, I reproduced it with Apache running on my Raspberry pi. This seems to be default behaviour of react-native, at least on Android. I posted a link to the relevant place in code in my reply to Tessus.

I'll add the comment as you suggested later today.

**Comment by laurent22** [comment]
Hmm, I saw the link you've posted but it's just setting up the cache size, which doesn't mean all requests are being cached by default? I would think it allocates the cache size if it's ever needed, but maybe I'm looking at it wrong.

If they enable caching by default, that doesn't sound like a smart thing to do and I'm surprised the issue only comes up now and only with Apache. But anyway, let's merge.

**Pull request merged**